src/dataflow-types/types.rs

```diff
@@ -458,7 +458,7 @@ impl ExternalSourceConnector {
 
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub enum Consistency {
-    BringYourOwn(String),
+    BringYourOwn(PathBuf),
```

Comment:
I'll revert to a string here as per your suggestion in your review. I'll make it explicit that we don't support non-utf8 paths in the conversation @benesch